bookserver/routers/assessment.py

```diff
@@ -70,38 +71,42 @@ async def get_assessment_results(
     request_data: AssessmentRequest,
     request: Request,
 ):
-    if not request.state.user:
+    user = request.state.user
+    if not user:
         return make_json_response(
             status=status.HTTP_401_UNAUTHORIZED, detail="not logged in"
         )
     # if the user is not logged in an HTTP 401 will be returned.
     # Otherwise if the user is an instructor then use the provided
-    # sid (it could be any student in the class) If none is provided then
+    # sid (it could be any student in the class). If none is provided then
     # use the user objects username
     if await is_instructor(request):
         if not request_data.sid:
-            request_data.sid = request.state.user.username
+            request_data.sid = user.username
     else:
         if request_data.sid:
             # someone is attempting to spoof the api
             return make_json_response(
                 status=status.HTTP_401_UNAUTHORIZED, detail="not an instructor"
             )
-        request_data.sid = request.state.user.username
+        request_data.sid = user.username
 
     row = await fetch_last_answer_table_entry(request_data)
     # mypy complains that ``row.id`` doesn't exist (true, but the return type wasn't exact and this does exist).
     if not row or row.id is None:  # type: ignore
         return make_json_response(detail="no data")
-
-    # :index:`todo``: **port the serverside grading** code::
-    #
-    #   do_server_feedback, feedback = is_server_feedback(div_id, course)
-    #   if do_server_feedback:
-    #       correct, res_update = fitb_feedback(rows.answer, feedback)
-    #       res.update(res_update)
-    rslogger.debug(f"Returning {row}")
-    return make_json_response(detail=row)
+    ret = row.dict()
+
+    # Do server-side grading if needed, which restores the answer and feedback.
+    if feedback := await is_server_feedback(request_data.div_id, request_data.course):
```

Comment:
Could a flag be provided as part of the AssessmentRequest that indicates if server side grading is used for this question?  -- This is the 3rd most called endpoint so I want to make sure we consider every additional database hit.

Reply:
Let me think about this -- good point.